btcpayserver: 1.0.7.2 -> 1.1.1

nixbitcoin opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

 You can run the tests with this [self-contained script](https://gist.github.com/nixbitcoin/b9a8b79eec0f39c684f94eb905bf68dc).

@erikarvstedt

nixbitcoin committed:
btcpayserver: 1.0.7.2 -> 1.1.1

erikarvstedt commented on pkgs/applications/blockchains/btcpayserver/default.nix:
```diff
@@ -15,13 +15,13 @@ in
 
 stdenv.mkDerivation rec {
   pname = "btcpayserver";
-  version = "1.0.7.0";
```
The deleted hunk must be `version = "1.0.7.2";`, otherwise the patch doesn't apply to master.

nixbitcoin replied:
My bad, forgot to fast-forward master

erikarvstedt commented:
[One extra commit](https://github.com/erikarvstedt/nixpkgs/commits/btcpayserver-1.1.0-ea).

erikarvstedt commented:
I'm getting [the following deps.nix content](https://github.com/erikarvstedt/nixpkgs/commit/15add50) when running `btcpayserver/update.sh`.
How did you update the deps?

nixbitcoin commented:
I'm running `./update.sh` from inside the `btcpayserver` directory.

r-rmcgibbo commented:
Result of `nixpkgs-review pr 121680` at 6518cbd8 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/2033ae51466d843186fd02e68135398d">btcpayserver</a></li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/applications/blockchains/btcpayserver/default.nix:29:3:
```
   |
29 |   buildPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/blockchains/btcpayserver/default.nix:43:3:
```
   |
43 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement.

nixbitcoin commented:
I found the bug, try with my newest HEAD

erikarvstedt committed:
nbxplorer: 2.1.49 -> 2.1.51

erikarvstedt committed:
btcpayserver/update.sh: auto-update nbxplorer

It's now sufficient to run a single script (btcpayserver/update.sh) to
update btcpayserver. This simplifies maintenance.
nbxplorer is a strict dependency of btcpayserver.

erikarvstedt commented:
Thanks for finding the bug! 
[Here's](https://github.com/erikarvstedt/nixpkgs/commits/btcpayserver-1.1.0-ea) a slightly better fix which enables relative path handling in `create-deps.sh`. Also, `nbxplorer: 2.1.49 -> 2.1.51`.

nixbitcoin commented:
Done, updated the [self-contained script](https://gist.github.com/nixbitcoin/b9a8b79eec0f39c684f94eb905bf68dc).

erikarvstedt reviewed (approved):
Things tested:

- `onlyCreateDeps=1 pkgs/applications/blockchains/btcpayserver/update.sh` produces the same content as this PR.
- The package builds and the nix-bitcoin VM test succeeds.

nixbitcoin commented:
Added another commit to significantly increase the supply chain integrity of this critical package.

nixbitcoin committed:
btcpayserver, nbxplorer: gpg verify upstream sources

erikarvstedt commented:
[Improve gpg verification](https://github.com/erikarvstedt/nixpkgs/commits/btcpayserver-1.1.0-ea-2):
- Fix that the git repo created in `update-common.sh` is never deleted because `trap` is overriden in `create-deps.sh`
- Remove unneeded tar roundtrip
- Only fetch the required release instead of the whole repo
- Add `refetch=1 ./update.sh` mode for reviewers, which re-verifies the content hash via GPG and recreates the deps file
- Use the new SRI hash format (`sha256-8aBv...`)

erikarvstedt commented:
I had forced-pushed some extra changes to [my branch](https://github.com/erikarvstedt/nixpkgs/commits/btcpayserver-1.1.0-ea-2), you might want to add them.

nixbitcoin commented:
Done

erikarvstedt commented:
btcpayserver 1.1.0 -> 1.1.1
[Fixed branch](https://github.com/erikarvstedt/nixpkgs/commits/btcpayserver-1.1.0-ea-3). This PR's title needs to be updated accordingly.

erikarvstedt reviewed (approved):
@SuperSandro2000
